### ISSUE TITLE
Kernel reports "Space between value and currency symbol = NO" for Germany 

### ISSUE THREAD

**Issue by bttrx** [FDOS/country#5]
This is when you use "!COUNTRY=049" in FDCONFIG.SYS, so no COUNTRY.SYS is involved.

But the kernel should report "YES" like in https://github.com/FDOS/country/blob/a9b2d3c1745e931d1a1dc9371bf3212ce6a8b519/country.asm#L3093.

So, please change "**1** , 2,_TIME_24" in https://github.com/FDOS/country/blob/a9b2d3c1745e931d1a1dc9371bf3212ce6a8b519/kernel.tb1#L71 to "**3** , 2,_TIME_24" to fix this.

**Comment by andrewbird** [comment]
You should be able to test the relevent kernel from here https://github.com/FDOS/kernel/actions/runs/1641887176
Thanks.

**Comment by bttrx** [comment]
I tested both new kernels.
Country data is returned correctly, except it reports country code = 1 (USA), although I set "country=049" in config.sys.

This seems to be an old issues, because I also found this problem in FD 1.1 and FD 1.2 kernels.
Had a quick look at the code, but have no idea so far:
https://github.com/FDOS/kernel/blob/202e3c1c94dbc3c268865ba21d18ed71a95524f1/kernel/inthndlr.c#L882
https://github.com/FDOS/kernel/blob/202e3c1c94dbc3c268865ba21d18ed71a95524f1/kernel/nls.c#L611
https://github.com/FDOS/kernel/blob/master/docs/nls.txt

Maybe you can spot it.

**Comment by andrewbird** [comment]
Hi Robert, 
Thanks for the report and info. Could you tell me which function or tool is reporting the incorrect value, please?

**Comment by bttrx** [comment]
Sure.
1. My very own: https://www.bttr-software.de/tmp/COUNTRY.ZIP
2. http://www.retroarchive.org/garbo/pc/sysinfo/ifp1p158.zip
3. Probably any other DOS sysinfo tool from: http://www.retroarchive.org/garbo/pc/sysinfo/index.html

**Comment by andrewbird** [comment]
Here's what I see under dosemu2 using your test program, current git kernel/country.sys, current git freecom and just adding `country=049` to fdconfig.sys

~~~
C:\>country
Country code       : 49
Date format        : 0001h (dd mm yy)
Time format        : 01h (Bit 0: 24-hour clock = YES)
Thousands separator: .
Decimal separator  : ,
Date separator     : .
Time separator     : :
Data-list separator: ,
Currency symbol    : €
Currency format    : 03h (Bit 0: Currency symbol follows value = YES
                          Bit 1: Space between value and currency symbol = YES
                          Bit 2: Currency symbol replaces decimal point = NO)
Currency precision : 2
Case map routine   : 00D9:12EE
C:\>ver/r
 
FreeCom version 0.85a - GNUC - XMS_Swap [Jan  2 2022 16:39:27]
DOS version 7.10
FreeDOS kernel - GIT (build 2043 OEM:0xfd) [compiled Jan  3 2022]    
~~~

I'll try the other programs, but I don't see anything wrong there, or did I miss something?

**Comment by andrewbird** [comment]
Infoplus 
![Screenshot_2022-01-03_13-07-13](https://user-images.githubusercontent.com/220835/147934380-224de347-4594-470f-843c-04f5351447c8.png)

**Comment by bttrx** [comment]
"current git kernel/country.sys" -> That's the culprit.
If kernel find's \country.sys during boot, kernel will make use of it, although you didn't say "...,a:\country.sys" in (fd)config.sys.
Solution is to rename or delete country.sys file in order to use the hard-coded kernel table.

**Comment by andrewbird** [comment]
Ahh, shows my lack of DOS knowledge. I'll try again later. Thanks!

**Comment by bttrx** [comment]
No problem. I also didn't know until 4 days ago, that there's a different way to provide (at least partial) country info.

**Comment by andrewbird** [comment]
BTW your country.exe program, is it open source? If so I'd like to include it (or some version of it if it's not C) into the dosemu2 test suite?

**Comment by bttrx** [comment]
Theoretically, it is open source... But it is just a little exercise done in an ancient Pascal variant (http://wiki.bttr-software.de/Cabezon/HomePage and https://github.com/cabezon-pascal/Cabezon).
That compiler required adding new functions directly to the RTL. So, w/o my modified RTL sources, country.exe source code wouldn't help you much. Unfortunately, I don't make any progress developing the compiler, because lack of time and other things.
But country.exe is a very simple program. Could be ported to Turbo Pascal or other languages easily. Comparing country.exe's output with Info+, I already see room for improvement.

**Comment by bttrx** [comment]
Any progress, @andrewbird ?

**Comment by andrewbird** [comment]
I'll try to have a look tomorrow.

**Comment by andrewbird** [comment]
> No problem. I also didn't know until 4 days ago, that there's a different way to provide (at least partial) country info.

So that's a FreeDOS only thing then, as I tried it on MSDOS 6.22 and DRDOS 7.01 and neither provided the fallback if no country.sys was present?

**Comment by bttrx** [comment]
> > No problem. I also didn't know until 4 days ago, that there's a different way to provide (at least partial) country info.
> 
> So that's a FreeDOS only thing then, as I tried it on MSDOS 6.22 and DRDOS 7.01 and neither provided the fallback if no country.sys was present?

I didn't try on other DOS, but I think, you're right.

**Comment by bttrx** [comment]
Sorry, to bother again, but any news, @andrewbird ?

**Comment by andrewbird** [comment]
Hello @bttrx,
   I don't know really. I did a quick fix that works okay for me, but after @stsp did a similar but different fix for FDPP I wasn't sure what to do about mine. I think I made the mistake of asking @PerditionC for thoughts in a comment to my commit https://github.com/andrewbird/kernel/commit/8751e33c345, but that commit subsequently got tweaked and so the comment became mostly inaccessible unless you know where to look.
